tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -203,14 +216,40 @@ ROOT::Experimental::Detail::RPageSinkDaos::CommitClusterImpl(ROOT::Experimental:
 void ROOT::Experimental::Detail::RPageSinkDaos::CommitDatasetImpl()
 {
    const auto &descriptor = fDescriptorBuilder.GetDescriptor();
-   auto szFooter = descriptor.GetFooterSize();
-   auto buffer = std::make_unique<unsigned char []>(szFooter);
-   descriptor.SerializeFooter(buffer.get());
-
-   auto zipBuffer = std::make_unique<unsigned char []>(szFooter);
-   auto szZipFooter = fCompressor->Zip(buffer.get(), szFooter, GetWriteOptions().GetCompression(),
-      [&zipBuffer](const void *b, size_t n, size_t o){ memcpy(zipBuffer.get() + o, b, n); } );
-   WriteNTupleFooter(zipBuffer.get(), szZipFooter, szFooter);
+
+   std::vector<DescriptorId_t> physClusterIDs;
+   for (const auto &c : descriptor.GetClusterIterable()) {
+      physClusterIDs.emplace_back(fSerializationContext.MapClusterId(c.GetId()));
+   }
+
+   auto szPageList =
+      Internal::RNTupleSerializer::SerializePageListV1(nullptr, descriptor, physClusterIDs, fSerializationContext);
+   auto bufPageList = std::make_unique<unsigned char[]>(szPageList);
+   Internal::RNTupleSerializer::SerializePageListV1(bufPageList.get(), descriptor, physClusterIDs,
+                                                    fSerializationContext);
+
+   auto bufPageListZip = std::make_unique<unsigned char[]>(szPageList);
+   auto szPageListZip = fCompressor->Zip(
+      bufPageList.get(), szPageList, GetWriteOptions().GetCompression(),
+      [&bufPageListZip](const void *b, size_t n, size_t o) { memcpy(bufPageListZip.get() + o, b, n); });
```

Comment:
This lambda "seems" repeated below.  Would it make sense (and clarify the code) to declare it only once (and give it a name)?

Reply:
Possibly, although the lambdas do differ in the variable that they capture.  This code part is anyway up for modification because we'll move to storing page lists (which correspond to cluster groups) not only at the very end but also along the way when writing large data sets.

I'll add a comment on the possible improvements on the lambdas.

Reply:
On second thought, it can be factored out in an (easy way)[https://github.com/root-project/root/pull/9483/commits/5b9a5c00f4fb98be535d82ea547b83617f13caca]

Reply:
Many thanks, @jblomer!